app/database_etl/test_adjustment_handler/test_adjustment.py

```diff
@@ -0,0 +1,246 @@
+    n_sp = fields.Integer()
+    y_sp = fields.Float()
+
+
+class TestAdjHandler:
+    def __init__(self, model_code=testadj_model_code, model_name='testadj_binomial_se_sp',
+                 execution_params={}):
+        self.TESTADJ_MODEL = self.get_stan_model_cache(model_code=model_code, model_name=model_name)
+        self.TESTADJ_MODEL_NAME = model_name
+        # Parse execution params
+        self.n_iter = execution_params.get('n_iter', 2000)
+        self.n_chains = execution_params.get('n_chains', 4)
+        self.return_fit = execution_params.get('return_fit', False)
+        self.n_replicates = execution_params.get('n_replicates', 5)
+        self.trials_lim = execution_params.get('trials_lim', 100)
+        self.modelsets_lim = execution_params.get('modelsets_lim', 3)
+
+    # make sure to gitignore model caches, because the model needs to be compiled separately
+    # on each machine - it is system-specific C++ code
+    def get_stan_model_cache(self, model_code: str, model_name: str = 'anon_model', **kwargs: Dict) -> pystan.StanModel:
+        """Use just as you would `pystan.StanModel`"""
+
+        # Create filepath of cached model
+        code_hash = md5(model_code.encode('ascii')).hexdigest()
+        cache_fn = f'stanmodelcache-{model_name}-{code_hash}.pkl'
+
+        # Try to load cached model
+        try:
+            cached_model = pickle.load(open(cache_fn, 'rb'))
+            print(f"Using cached StanModel at filepath {cache_fn}")
+
+        # Otherwise create the model and cache it
+        except FileNotFoundError:
+            cached_model = pystan.StanModel(model_code=model_code, model_name=model_name, **kwargs)
+            with open(cache_fn, 'wb') as f:
+                pickle.dump(cached_model, f)
+        return cached_model
+
+    def fit_one_pystan_model(self, model_params: Dict) -> Tuple:
+        fit = self.TESTADJ_MODEL.sampling(data=model_params,
+                                          iter=self.n_iter,
+                                          chains=self.n_chains,
+                                          control={'adapt_delta': 0.95},
+                                          check_hmc_diagnostics=False)
+
+        summary = fit.summary()
+        summary_df = pd.DataFrame(data=summary['summary'],
+                                  index=summary['summary_rownames'],
+                                  columns=summary['summary_colnames'])
+
+        samples = fit.extract(pars='prev', permuted=True)['prev']
+
+        summary_df_parsed = summary_df.loc['prev', ['50%', '2.5%', '97.5%']]
+        summary_df_parsed['samples'] = samples
+        summary_df_parsed['fit'] = fit
+
+        diagnostics = pystan.diagnostics.check_hmc_diagnostics(fit)
+
+        satisfactory_model_found = all(diagnostics.values())
+        return summary_df_parsed, satisfactory_model_found
+
+    def pystan_adjust(self, model_params: Dict, execution_params: Dict = {}) -> Union[Tuple, pystan.StanModel]:
```

Comment:
I think we can remove execution_params here because we deal with them in init.

Also we can add credible_interval_size as a class property (see how we deal with n_replicates) and remove the get statement from here